engine/class_modules/sc_shaman.cpp

```diff
@@ -5303,6 +5291,13 @@ struct spiritwalkers_grace_t : public shaman_spell_t
                       options_str )
   {
     may_miss = may_crit = harmful = callbacks = false;
+    // Reduces the cooldown of Spiritwalker's Grace by ${$m1/-1000} sec
+    if ( p()->talent.graceful_spirit->ok() )
+    {
+      timespan_t reduction = timespan_t::from_seconds( p()->talent.graceful_spirit->effectN( 1 ).base_value() / -1000 );
+      cooldown->duration -= reduction;
```

Comment:
Or make this a one liner:
`cooldown->duration += p()->talent.graceful_spirit->effectN( 1 ).time_value();`

"+" there because the time value is negative.